clio-client/src/main/scala/org/broadinstitute/clio/client/metadata/ArrayDeliverer.scala

```diff
@@ -40,12 +40,18 @@ class ArrayDeliverer extends MetadataMover[ArraysMetadata] {
       redIdatPath = src.redIdatPath.map(buildFilePath(_, idatDestination))
     )
 
+    // If the Array has already been delivered, we want to move the idats instead of copying them.
+    val idatOp = if (src.workspaceName.forall(_.isEmpty)) {
```

Comment:
Phil explained that there is a idat copy per array delivery in FireCloud.
Idat files are not shared across arrays there.

Reply:
Since we're only doing this for arrays delivery and for this specific case, I don't see why we need to abstract the logic out.

Reply:
It looks like it could be done in `DeliverableMetadata` and then shared with the code in `CramMetadata.pathsToDelete`. There is a similar use in `DeliverExecutor.checkPreconditions` but that is testing to see if it has been delivered and the workspace name is different.